components/nodemanager-service/api/nodes/server/server.go

```diff
@@ -264,8 +266,12 @@ func (srv *Server) Delete(ctx context.Context, in *nodes.Id) (*pb.Empty, error)
 
 // List nodes based on a query
 func (srv *Server) List(ctx context.Context, in *nodes.Query) (*nodes.Nodes, error) {
-	logrus.Debugf("Getting Nodes with query: %+v", in)
-	dbnodes, totalCount, err := srv.db.GetNodes(in.Sort, in.Order, in.Page, in.PerPage, in.Filters)
+	filters, err := addProjectFilters(ctx, in.Filters)
```

Comment:
Getting the project filters from the context and adding them to the query's filters.